solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,250 @@
+    ///      transfer.
+    /// @param from TBTC v1 token holder minting TBTC v2 tokens
+    /// @param amount The amount of TBTC v2 to mint from TBTC v1
+    /// @param token TBTC v1 token address
+    function receiveApproval(
+        address from,
+        uint256 amount,
+        address token,
+        bytes calldata
+    ) external override {
+        require(token == address(tbtcV1), "Token is not TBTC v1");
+        require(msg.sender == address(tbtcV1), "Only TBTC v1 caller allowed");
+        _mint(from, amount);
+    }
+
+    /// @notice Unmints TBTC v2 from the caller into TBTC v1. Depending on
+    ///         `unmintFee` value, may require paying an additional unmint fee
+    ///          in TBTC v2 in addition to the amount being unminted. To see
+    ///          what is the value of the fee, please call `unmintFeeFor(amount)`
+    ///          function. The caller needs to have at least
+    ///          `amount + unmintFeeFor(amount)` of TBTC v2 balance approved for
+    ///          transfer to the `VendingMachine` before calling this function.
+    /// @param amount The amount of TBTC v2 to unmint to TBTC v1
+    function unmint(uint256 amount) external {
+        uint256 fee = unmintFeeFor(amount);
+        emit Unminted(msg.sender, amount, fee);
+
+        require(
+            tbtcV2.balanceOf(msg.sender) >= amount + fee,
+            "Amount + fee exceeds TBTC v2 balance"
+        );
+
+        tbtcV2.safeTransferFrom(msg.sender, address(this), fee);
+        tbtcV2.burnFrom(msg.sender, amount);
+        tbtcV1.safeTransfer(msg.sender, amount);
+    }
+
+    /// @notice Allows the Governance to withdraw unmint fees accumulated by
+    ///         `VendingMachine`.
+    /// @param recipient The address receiving the fees
+    /// @param amount The amount of fees in TBTC v2 to withdraw
+    function withdrawFees(address recipient, uint256 amount)
+        external
+        onlyOwner
+    {
+        tbtcV2.safeTransfer(recipient, amount);
+    }
+
+    /// @notice Allows the Governance to begin unmint fee update process.
+    ///         The update process needs to be finalized with a call to
+    ///         `finalizeUnmintFeeUpdate` function after the `GOVERNANCE_DELAY`
+    ///         passes.
+    /// @param _newUnmintFee The new unmint fee
+    function beginUnmintFeeUpdate(uint256 _newUnmintFee) external onlyOwner {
+        /* solhint-disable-next-line not-rely-on-time */
+        emit UnmintFeeUpdateStarted(_newUnmintFee, block.timestamp);
+        newUnmintFee = _newUnmintFee;
+        /* solhint-disable-next-line not-rely-on-time */
+        unmintFeeChangeInitiated = block.timestamp;
+    }
+
+    /// @notice Allows the Governance to finalize unmint fee update process.
+    ///         The update process needs to be first initiated with a call to
+    ///         `beginUnmintFeeUpdate` and the `GOVERNANCE_DELAY` needs to pass.
+    function finalizeUnmintFeeUpdate()
+        external
+        onlyOwner
```

Comment:
Is the `onlyOwner` modifier necessary here? Can't the finalization be callable by anyone?

Reply:
All `finalize*` calls in tBTC are `onlyOwner`, same for coverage pools so I decided to follow the same pattern here.